index.js

```diff
@@ -7,22 +7,24 @@ const { PassThrough } = require('readable-stream')
 const glob = require('glob')
 const send = require('send')
 const fp = require('fastify-plugin')
+const util = require('util')
+const globPromise = util.promisify(glob)
 
 const dirList = require('./lib/dirList')
 
-function fastifyStatic (fastify, opts, next) {
+async function fastifyStatic (fastify, opts) {
   const error = checkRootPathForErrors(fastify, opts.root)
-  if (error !== undefined) return next(error)
+  if (error !== undefined) throw error
```

Comment:
minor. this pattern of returning an error was necessary in the callback based version but it's no longer necessary now. it could be changed so that the `checkRootPathForErrors` function immediately throws an error, meaning that if that function is invoked and no error is thrown, then the plugin registration can continue and there would be no need, here, to test the return value of that function. in any case, I appreciate this would require more changes so it's also ok if you leave it as it is.

Reply:
`checkRootPathForErrors` now immediately throws on any error with the root path option.